### PR TITLE
Add request + response packet to request rest endpoint

### LINKED ISSUE #1882: Add request + response packet to request rest endpoint
After we changed how to store the result in store, the endpoint can show result (request and response packet).

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1882

### CODE REVIEW

File: chain/x/oracle/types/querier.go [hunk 1]
```diff
@@ -17,9 +17,15 @@ type QueryCountsResult struct {
...
+	RequestPacketData  OracleRequestPacketData
+	ResponsePacketData OracleResponsePacketData
+}
+
 // QueryRequestResult is the struct for the result of request query.
 type QueryRequestResult struct {
 	Request Request  `json:"request"`
 	Reports []Report `json:"reports"`
-	Result  []byte   `json:"result"` // TODO: Update to richer data type.
+	Result  Result   `json:"result"`
```
Comment (sorawit) [inline chain/x/oracle/types/querier.go]: Make this pointer.

File: chain/x/oracle/keeper/querier.go [hunk 2]
```diff
@@ -95,7 +96,9 @@ func queryRequestByID(ctx sdk.Context, path []string, k Keeper) ([]byte, error)
 		return nil, err
 	}
 	reports := k.GetReports(ctx, types.RequestID(id))
-	result, _ := k.GetResult(ctx, types.RequestID(id))
+	resultBytes, _ := k.GetResult(ctx, types.RequestID(id))
```
Comment (sorawit) [inline chain/x/oracle/keeper/querier.go]: GetResult return pair type. MustGetResult.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/client/common/request_search.go [hunk 3]
```diff
@@ -52,8 +52,15 @@ func QuerySearchLatestRequest(
 					}
 				}
 				if ok && rid != "" {
-					// TODO: Check if request has resolved.
-					return queryRequest(route, cliCtx, rid)
+					res, h, err := queryRequest(route, cliCtx, rid)
+					if err != nil {
```
Comment (sorawit) [inline chain/x/oracle/client/common/request_search.go:56]: if err != nil we should just return error in my opinion. We never expect to get error.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -32,6 +33,7 @@
 - (imprv) [\#1904](https://github.com/bandprotocol/bandchain/pull/1904) Added helper tooltips.
 - (imprv) [\#1900](https://github.com/bandprotocol/bandchain/pull/1900) Shorten marketcap amount on landing page
 - (feat) [\#1888](https://github.com/bandprotocol/bandchain/pull/1888) Added OBI bindings and patched scan to use OBI standard
+- (impv) [\#1900](https://github.com/bandprotocol/bandchain/pull/1900) Shorten marketcap amount on landing page
```
Comment (sorawit) [inline CHANGELOG.md]: let's remove this?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/querier.go [hunk 5]
```diff
@@ -95,10 +95,10 @@ func queryRequestByID(ctx sdk.Context, path []string, k Keeper) ([]byte, error)
 		return nil, err
 	}
 	reports := k.GetReports(ctx, types.RequestID(id))
-	result, _ := k.GetResult(ctx, types.RequestID(id))
+	result := k.MustGetResult(ctx, types.RequestID(id))
```
Comment (taobun) [inline chain/x/oracle/keeper/querier.go:107]: I think it should use HasResult first if the result doesn't exist it should return nil not panic endpoint.

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/result.go [hunk 6]
```diff
@@ -19,12 +20,26 @@ func (k Keeper) SetResult(ctx sdk.Context, reqID types.RequestID, result []byte)
...
+	var result types.Result
+	err := obi.Decode(bz, &result)
+	if err != nil {
+		return types.Result{}, types.ErrOBIDecode
+	}
+
+	return result, nil
+}
+
+// MustGetResult returns the result bytes for the given request ID. Panics on error.
```
Comment (taobun) [inline chain/x/oracle/keeper/result.go]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/querier.go [hunk 7]
```diff
@@ -95,10 +95,19 @@ func queryRequestByID(ctx sdk.Context, path []string, k Keeper) ([]byte, error)
 		return nil, err
 	}
 	reports := k.GetReports(ctx, types.RequestID(id))
-	result, _ := k.GetResult(ctx, types.RequestID(id))
+
+	if !k.HasRequest(ctx, types.RequestID(id)) {
```
Comment (taobun) [inline chain/x/oracle/keeper/querier.go]: HasResult!!

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**